deltacat/tests/compute/test_compact_partition_multiple_rounds.py

```diff
@@ -327,4 +328,5 @@ def test_compact_partition_rebase_multiple_rounds_same_source_and_destination(
     if assert_compaction_audit:
         if not assert_compaction_audit(compactor_version, compaction_audit):
             assert False, "Compaction audit assertion failed"
+    assert object_store_delete_many_spy.call_count, "Object store was never cleaned up!"
```

Comment:
This just checks if delete_many was called. We want to test behavior i.e., the object store actually ends up with no data after the compaction.